index-handlers/src/main/java/no/sikt/nva/nvi/index/aws/OpenSearchClient.java

```diff
@@ -113,6 +115,30 @@ public void deleteIndex() throws IOException {
             .delete(new DeleteIndexRequest.Builder().index(NVI_CANDIDATES_INDEX).build());
     }
 
+    //TODO change with .exists() when sws index handler is cleaned up.
+    public boolean indexExists() {
+        try {
+            client.withTransportOptions(getOptions())
+                .indices()
+                .get(GetIndexRequest.of(r -> r.index(NVI_CANDIDATES_INDEX)));
+        } catch (IOException io) {
+            throw new RuntimeException(io);
+        } catch (OpenSearchException osex) {
+            if (osex.status() == 404 && INDEX_NOT_FOUND_EXCEPTION.equals(osex.error().type())) {
+                return false;
+            }
+            throw osex;
+        }
+        return true;
+    }
```

Comment:
Remove it from future implementation of indexing? No need to do this call for every nvi candidate.

Reply:
It's not called for every update anymore, see [Fix: NVI Search endpoint returns 500 Internal Server Error if no candidates exists #NP-45369 (](https://github.com/BIBSYSDEV/nva-nvi/commit/7b232cc83208d5adae2692b3b0c51f5d31dcbe76)https://github.com/BIBSYSDEV/nva-nvi/pull/177[)](https://github.com/BIBSYSDEV/nva-nvi/commit/7b232cc83208d5adae2692b3b0c51f5d31dcbe76)